app/master/build.py

```diff
@@ -326,6 +340,10 @@ def cancel(self):
         self._state_machine.trigger(BuildEvent.CANCEL)
 
     def _on_enter_canceled_state(self, event):
+        # Set the kill_event to kill the subprocesses for the build
+        self._logger.notice('Set the kill_event for subprocesses for build {}.', self._build_id)
+        self.project_type.kill_subprocesses()
+
```

Comment:
Was there a problem with subprocesses not being killed? 

Is the bottom block for draining the unstarted_subjobs queue still necessary?